spring-kafka/src/main/java/org/springframework/kafka/retrytopic/ExceptionBasedDltDestination.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2016-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.retrytopic;
+
+/**
+ * Annotation allowing to specify additional DLT which will be chosen when message
+ * processing caused the configured exception to be thrown.
+ *
+ * @author Adrian Chlebosz
+ * @see org.springframework.kafka.retrytopic.ExceptionBasedDltRouting
+ * @since 3.2.0
```

Comment:
I think these two has to be swapped according to the Checkstyle rules:
```
		<module name="AtclauseOrder">
			<property name="target" value="CLASS_DEF, INTERFACE_DEF, ENUM_DEF"/>
			<property name="tagOrder" value="@param, @author, @since, @see, @version, @serial, @deprecated"/>
		</module>
```

Also it would be great to have an `@author` in its own paragraph to really make it visible for an honor.